test/integration/datapath/dapapath_linux_test.go

```diff
@@ -0,0 +1,325 @@
+   or
+go test -count=1 test/integration/datapath/datapath_linux_test.go -timeout 3m -tags connection -run ^TestDatapathLinux$ -podName=acnpod -nodepoolSelector=aks-pool1 -tags=connection,integration
+
+
+This test checks pod to pod, pod to node, pod to Internet check
+
+Timeout context is controled by the -timeout flag.
+
+*/
+
+// return podLabelSelector and nodeLabelSelector
+func createLabelSelectors() (string, string) {
+	return fmt.Sprintf("%s=%s", podLabelKey, *podPrefix), fmt.Sprintf("%s=%s", nodepoolKey, *nodepoolSelector)
+}
+
+func setupLinuxEnvironment(t *testing.T) {
+	ctx := context.Background()
+
+	t.Log("Create Clientset")
+	clientset, err := k8sutils.MustGetClientset()
+	if err != nil {
+		require.NoError(t, err, "could not get k8s clientset: %v", err)
+	}
+
+	t.Log("Create Label Selectors")
+	podLabelSelector, nodeLabelSelector := createLabelSelectors()
+
+	t.Log("Get Nodes")
+	nodes, err := k8sutils.GetNodeListByLabelSelector(ctx, clientset, nodeLabelSelector)
+	if err != nil {
+		require.NoError(t, err, "could not get k8s node list: %v", err)
+	}
+
+	createPodFlag := !(apierrors.IsAlreadyExists(err))
+	t.Logf("%v", createPodFlag)
+
+	if createPodFlag {
+		var daemonset appsv1.DaemonSet
+		t.Log("Creating Linux pods through deployment")
+		deployment, err := k8sutils.MustParseDeployment(*testProfile)
+		if err != nil {
+			require.NoError(t, err)
+		}
+
+		if *testProfile == LinuxDeployIPV4 {
+			daemonset, err = k8sutils.MustParseDaemonSet(gpDaemonset)
+			if err != nil {
+				t.Fatal(err)
+			}
+		} else {
+			daemonset, err = k8sutils.MustParseDaemonSet(gpDaemonsetIPv6)
+			if err != nil {
+				t.Fatal(err)
+			}
+		}
+
+		rbacCleanUpFn, err := k8sutils.MustSetUpClusterRBAC(ctx, clientset, gpClusterRolePath, gpClusterRoleBindingPath, gpServiceAccountPath)
+		if err != nil {
+			t.Log(os.Getwd())
+			t.Fatal(err)
+		}
+
+		// Fields for overwritting existing deployment yaml.
+		// Defaults from flags will not change anything
+		deployment.Spec.Selector.MatchLabels[podLabelKey] = *podPrefix
+		deployment.Spec.Template.ObjectMeta.Labels[podLabelKey] = *podPrefix
+		deployment.Spec.Template.Spec.NodeSelector[nodepoolKey] = *nodepoolSelector
+		deployment.Name = *podPrefix
+		deployment.Namespace = *podNamespace
+
+		deploymentsClient := clientset.AppsV1().Deployments(*podNamespace)
+		err = k8sutils.MustCreateDeployment(ctx, deploymentsClient, deployment)
+		if err != nil {
+			require.NoError(t, err)
+		}
+
+		daemonsetClient := clientset.AppsV1().DaemonSets(daemonset.Namespace)
+		err = k8sutils.MustCreateDaemonset(ctx, daemonsetClient, daemonset)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		t.Cleanup(func() {
+			t.Log("cleaning up resources")
+			rbacCleanUpFn()
+
+			if err := deploymentsClient.Delete(ctx, deployment.Name, metav1.DeleteOptions{}); err != nil {
+				t.Log(err)
+			}
+
+			if err := daemonsetClient.Delete(ctx, daemonset.Name, metav1.DeleteOptions{}); err != nil {
+				t.Log(err)
+			}
+		})
```

Comment:
Nit: Lets get the relevant code together where possible. Parse, define, create together for deployment and daemonset respectiviely.